redis/tests/test_async.rs

```diff
@@ -638,6 +638,23 @@ mod basic_async {
         );
     }
 
+    #[tokio::test]
+    async fn test_scan_with_options_works() {
+        let ctx = TestContext::new();
+        let mut con = ctx.multiplexed_async_connection().await.unwrap();
+        for i in 0..20usize {
+            let _: () = con.append(format!("test/{i}"), i).await.unwrap();
```

Comment:
please also add values that don't have `test/` prefix, in order to verify that only `test/` values are returned, and then also test that without setting `pattern` they are returned.